### PR TITLE
Add EntraID support for Azure Cache for Redis

### PR DESCRIPTION
**Pull request by hhunter-ms**
## Description

Add `useEntraID` metadata
Update Setup/Create Redis sections for Azure to be consistent and include instructions for `useEntraID`

## Issue reference

PR will close: #4120 

## Related links

dapr/components-contrib#3470

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-bindings/redis.md [hunk 1]
```diff
@@ -160,7 +163,7 @@ Dapr can use any Redis instance - containerized, running on your local dev machi
 
 *Note: Dapr does not support Redis >= 7. It is recommended to use Redis 6*
 
-{{< tabs "Self-Hosted" "Kubernetes" "AWS" "GCP" "Azure">}}
+{{< tabs "Self-Hosted" "Kubernetes" "AWS" "GCP" "Azure Cache for Redis">}}
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/redis.md]: If you are using the full product name for Azure, then we should do the same for AWS (https://docs.aws.amazon.com/whitepapers/latest/scale-performance-elasticache/elasticache-for-redis.html) and GCP. Do you want to have the full product name?

File: daprdocs/content/en/reference/components-reference/supported-configuration-stores/redis-configuration-store.md [hunk 2]
```diff
@@ -38,32 +39,33 @@ The above example uses secrets as plain strings. It is recommended to use a secr
...
+| minIdleConns        | N | Minimum number of idle connections to keep open in order to avoid the performance degradation associated with creating new connections. Defaults to `"0"`. | `"2"`
+| idleCheckFrequency        | N | Frequency of idle checks made by idle connections reaper. Default is `"1m"`. `"-1"` disables idle connections reaper. | `"-1"`
+| idleTimeout        | N | Amount of time after which the client closes idle connections. Should be less than server's timeout. Default is `"5m"`. `"-1"` disables idle timeout check. | `"10m"`
 
 ## Setup Redis
 
 Dapr can use any Redis instance: containerized, running on your local dev machine, or a managed cloud service.
 
-{{< tabs "Self-Hosted" "Kubernetes" "Azure" "AWS" "GCP" >}}
+{{< tabs "Self-Hosted" "Kubernetes" "Azure Cache for Redis" "AWS" "GCP" >}}
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-configuration-stores/redis-configuration-store.md]: Same comment here on product naming. Also the order is different here. Suggest AWS, Azure and GCP (alphabetical). If so would also change above ^^

File: daprdocs/content/en/reference/components-reference/supported-locks/redis-lock.md [hunk 3]
```diff
@@ -108,7 +111,7 @@ The above example uses secrets as plain strings. It is recommended to use a secr
 
 Dapr can use any Redis instance: containerized, running on your local dev machine, or a managed cloud service.
 
-{{< tabs "Self-Hosted" "Kubernetes" "Azure" "AWS" "GCP" >}}
+{{< tabs "Self-Hosted" "Kubernetes" "Azure Cache for Redis" "AWS" "GCP" >}}
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-locks/redis-lock.md]: Same comment here on product name for all the clouds

**Review by msfussell — COMMENTED** [review]
A few comments

File: daprdocs/content/en/reference/components-reference/supported-bindings/redis.md (hunk 1 above)
Comment (hhunter-ms) [reply daprdocs/content/en/reference/components-reference/supported-bindings/redis.md]: I'll leave it as Azure, but will rearrange to be alphabetical

**Review by msfussell — APPROVED** [review]
LGTM

**Pull request merged**